src/Tizen.Applications.Common/Tizen.Applications/Event.cs

```diff
@@ -0,0 +1,134 @@
+﻿/*
+ * Copyright (c) 2018 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using static Interop.Event;
+
+namespace Tizen.Applications
+{
+    /// <summary>
+    /// The Event API provides functions to receive events.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class Event : IDisposable
+    {
+        internal IntPtr _handler;
+        private readonly string _eventName = null;
+        private Interop.Event.EventCallback _eventCallBack;
+
+        /// <summary>
+        /// Initializes the instance of the Event class.
+        /// </summary>
+        /// <param name="eventName">The name of the event.</param>
+        /// <exception cref="System.ArgumentException">Thrown when eventName is null or empty.</exception>
+        /// <example>
+        /// <code>
+        /// Event ev = new Event("tizen.system.event.usb_status");
+        /// ev += EventReceivedCallback;
+        /// </code>
+        /// </example>
+        /// <since_tizen> 6 </since_tizen>
+        public Event(string eventName)
+        {
+            if (String.IsNullOrEmpty(eventName))
+            {
+                throw new InvalidOperationException("null event name");
+            }
+            _eventName = eventName;
+            _eventCallBack = (string name, IntPtr eventData, IntPtr userData) =>
+            {
+                ApplicationEventArgs args = new ApplicationEventArgs()
+                {
+                    EventData = new Bundle(new SafeBundleHandle(eventData, false))
```

Comment:
Why not has ownership of Bundle?
It has a issue lifecycle of `EventData`

Q1. eventData was valid on outside of this callback?

Q1-1. if valid outside of this callback, 
   Who is release evetData and when is released? If anyone dose not release it, 
    new created Bundle object shoud has owner ship of this eventData
Q1-2. if invalid outside of this callback.
  You should copy a eventData to make sure valid on outside of this callback.